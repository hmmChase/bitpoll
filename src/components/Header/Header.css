header {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  background: #cbccdb;
  justify-content: space-around;
  width: 100%;
}

h1 {
  font-family: 'Ubuntu', serif;
  font-weight: 800px;
  font-style: italic;
  color: #4d4d4d;
}

.logo-icon {
  color: #f7931a;
}

.signed-in-state {
  display: flex;
  align-items: center;
  margin: 20px;
}

.sign-in-btn {
  background: #3a3133;
  color: #fff;
  border: none;
  border-radius: 10px;
  height: 50px;
  display: flex;
  align-items: center;
  padding: 10px;
}

.sign-out-btn {
  background: #3a3133;
  color: #fff;
  border: none;
  border-radius: 10px;
  height: 50px;
  display: flex;
  align-items: center;
  padding: 10px;
  margin-left: 20px;
}

.sign-in-btn:hover {
  background: #000;
}

.sign-in-btn-txt {
  margin-left: 5px;
}

.log-in-out {
  position: relative;
}

.welcome-text {
  color: #000;
}

.verified-icon {
  margin-right: 2px;
}

.verifed-text {
  font-size: 0.7em;
  position: absolute;
  top: 25px;
  color: green;
}
