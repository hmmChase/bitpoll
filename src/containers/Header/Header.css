header {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  background: #cbccdb;
  justify-content: space-around;
  width: 100%;
}

h1 {
  font-family: 'Ubuntu', serif;
  font-weight: 800px;
  font-style: italic;
  color: #4d4d4d;
  margin: 10px;
}

.logo-icon {
  color: #f7931a;
}
