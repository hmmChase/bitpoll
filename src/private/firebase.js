export const config = {
  apiKey: 'AIzaSyBtlh3EWl68895pmLKVoHpw95G2QYKsYf8',
  authDomain: 'bit-poll.firebaseapp.com',
  databaseURL: 'https://bit-poll.firebaseio.com',
  projectId: 'bit-poll',
  storageBucket: 'bit-poll.appspot.com',
  messagingSenderId: '1009809453468'
};
