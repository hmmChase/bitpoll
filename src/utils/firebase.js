import firebase from 'firebase';
import { config } from '../private/firebase';

firebase.initializeApp(config);

export const provider = new firebase.auth.GithubAuthProvider();
export const auth = firebase.auth();

export default firebase;
