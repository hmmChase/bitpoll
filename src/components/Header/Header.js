import React, { Component } from 'react';
import { connect } from 'react-redux';
import { auth, provider } from '../../utils/firebase';
import fontawesome from '@fortawesome/fontawesome';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import faGithub from '@fortawesome/fontawesome-free-brands';
import faCheck from '@fortawesome/fontawesome-free-regular';
import * as actions from '../../actions/index';
import './Header.css';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  componentDidMount() {
    this.refreshLogin();
  }

  refreshLogin = () => {
    auth.onAuthStateChanged(user => {
      if (user) {
        this.setLogin(user);
      }
    });
  };

  login = async () => {
    const user = await this.authGitHub();
    this.setLogin(user);
  };

  authGitHub = () => {
    return auth.signInWithPopup(provider).then(result => {
      return result.user;
    });
  };

  setLogin = user => {
    const cleanUser = this.cleanUser(user);
    this.props.storeUser(cleanUser);
    this.props.getContributors();
  };

  cleanUser = user => ({
    userId: user.providerData[0].uid,
    displayName: user.providerData[0].displayName
  });

  logOut = () => {
    auth.signOut().then(() => {
      this.props.storeLogOut();
    });
  };

  render() {
    return (
      <header>
        <div>
          <h1 className="App-title">
            <FontAwesomeIcon
              className="logo-icon"
              icon={['far', 'check-circle']}
            />
            Bitpoll
          </h1>
        </div>

        {this.props.userId ? (
          <div className="signed-in-state">
            <div className="log-in-out">
              <p className="welcome-text">Welcome {this.props.displayName}</p>
              {this.props.isContributor && (
                <p className="verifed-text">
                  <FontAwesomeIcon icon={['far', 'check-circle']} />verified
                  contributor
                </p>
              )}
            </div>

            <div className="sign-out-btn" onClick={this.logOut}>
              <span>Log out</span>
            </div>
          </div>
        ) : (
          <div className="sign-in-btn" onClick={this.login}>
            <FontAwesomeIcon icon={['fab', 'github']} />
            <span className="sign-in-btn-txt">Sign in with GitHub</span>
          </div>
        )}
      </header>
    );
  }
}

export const mapStateToProps = state => ({
  displayName: state.user.displayName,
  userId: state.user.userId,
  isContributor: state.user.isContributor,
  contributors: state.contributors
});

export const mapDispatchToProps = dispatch => ({
  storeUser: user => dispatch(actions.storeUser(user)),
  storeLogOut: () => dispatch(actions.storeLogOut()),
  getContributors: () =>
    dispatch(
      actions.getContributors(
        'https://api.github.com/repos/bitcoin/bitcoin/contributors?per_page=100'
      )
    )
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
