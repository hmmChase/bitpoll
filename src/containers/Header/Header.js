import React, { Component } from 'react';
import { connect } from 'react-redux';
import { auth, provider } from '../../utils/firebase';
import SignIn from './SignIn/SignIn';
import SignedIn from './SignedIn/SignedIn';
import fontawesome from '@fortawesome/fontawesome';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import faGithub from '@fortawesome/fontawesome-free-brands';
import faCheck from '@fortawesome/fontawesome-free-regular';
import * as actions from '../../actions/index';
import './Header.css';

export class Header extends Component {
  componentDidMount() {
    this.refreshLogin();
  }

  refreshLogin = () => {
    auth.onAuthStateChanged(user => {
      if (user) {
        this.setLogin(user);
      }
    });
  };

  login = async () => {
    const user = await this.authGitHub();
    this.setLogin(user);
  };

  authGitHub = async () => {
    const result = await auth.signInWithPopup(provider);
    return result.user;
  };

  setLogin = user => {
    const cleanUser = this.cleanUser(user);
    this.props.storeUser(cleanUser);
    this.props.getContributors();
  };

  cleanUser = user => ({
    userId: user.providerData[0].uid,
    displayName: user.providerData[0].displayName
  });

  logOut = async () => {
    await auth.signOut();
    this.props.storeLogOut();
  };

  render() {
    return (
      <header>
        <h1 className="Header-title">
          <FontAwesomeIcon
            className="logo-icon"
            icon={['far', 'check-circle']}
          />
          Bitpoll
        </h1>
        {this.props.userId ? (
          <SignedIn logOut={this.logOut} />
        ) : (
          <SignIn login={this.login} />
        )}
      </header>
    );
  }
}

export const mapStateToProps = state => ({
  userId: state.user.userId
});

export const mapDispatchToProps = dispatch => ({
  storeUser: user => dispatch(actions.storeUser(user)),
  storeLogOut: () => dispatch(actions.storeLogOut()),
  getContributors: () =>
    dispatch(
      actions.getContributors(
        'https://api.github.com/repos/bitcoin/bitcoin/contributors?per_page=100'
      )
    )
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Header);
