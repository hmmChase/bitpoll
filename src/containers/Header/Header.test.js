import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './Header';
import { auth, provider } from '../../utils/firebase';
jest.mock('../../utils/firebase.js');

describe('Header', () => {
  let header;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      contributors: {},
      displayName: 'Chase',
      getContributors: jest.fn(),
      isContributor: false,
      storeLogOut: jest.fn(),
      storeUser: jest.fn(),
      userId: '10532835'
    };
    header = shallow(<Header {...mockProps} />, {
      disableLifecycleMethods: true
    });
  });

  it('matches the snapshot', () => {
    expect(header).toMatchSnapshot();
  });

  describe('componentDidMount', () => {
    it('calls refreshLogin', () => {
      const refreshLogin = (header.instance().refreshLogin = jest.fn());
      header.instance().componentDidMount();

      expect(refreshLogin).toHaveBeenCalledTimes(1);
    });
  });

  describe('refreshLogin', () => {
    it('calls auth.onAuthStateChanged', () => {
      header.instance().refreshLogin();

      expect(auth.onAuthStateChanged).toHaveBeenCalledTimes(1);
    });

    // it('calls setLogin', () => {
    //   const user = { id: 1111 };

    //   auth.onAuthStateChanged(user);

    //   const setLogin = (header.instance().setLogin = jest.fn());
    //   header.instance().refreshLogin();

    //   expect(setLogin).toHaveBeenCalledTimes(1);
    // });
  });

  describe('login', () => {
    it.only('calls authGitHub', async () => {
      console.log('asdf');

      const authGitHub = (header.instance().authGitHub = jest.fn());
      authGitHub.mockImplementation(() => Promise.resolve({}));

      console.log('hi');

      console.log('authGitHub: ', authGitHub);

      await header.instance().login();

      console.log('bye');

      console.log('authGitHub: ', authGitHub);

      await expect(await authGitHub).toHaveBeenCalledTimes(1);
    });
  });
});
