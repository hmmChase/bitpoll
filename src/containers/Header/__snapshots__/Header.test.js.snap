// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot with no userId set 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    contributors={Object {}}
    displayName="Chase"
    getContributors={[Function]}
    isContributor={false}
    storeLogOut={[Function]}
    storeUser={[Function]}
    userId=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="Header-title"
>
          <FontAwesomeIcon
                    border={false}
                    className="logo-icon"
                    fixedWidth={false}
                    flip={null}
                    icon={
                              Array [
                                        "far",
                                        "check-circle",
                                      ]
                    }
                    listItem={false}
                    mask={null}
                    name=""
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    transform={null}
          />
          Bitpoll
</h1>,
        <SignIn
          login={[Function]}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <FontAwesomeIcon
              border={false}
              className="logo-icon"
              fixedWidth={false}
              flip={null}
              icon={
                            Array [
                                          "far",
                                          "check-circle",
                                        ]
              }
              listItem={false}
              mask={null}
              name=""
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              transform={null}
/>,
            "Bitpoll",
          ],
          "className": "Header-title",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "border": false,
              "className": "logo-icon",
              "fixedWidth": false,
              "flip": null,
              "icon": Array [
                "far",
                "check-circle",
              ],
              "listItem": false,
              "mask": null,
              "name": "",
              "pull": null,
              "pulse": false,
              "rotation": null,
              "size": null,
              "spin": false,
              "symbol": false,
              "transform": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "Bitpoll",
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "login": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="Header-title"
>
            <FontAwesomeIcon
                        border={false}
                        className="logo-icon"
                        fixedWidth={false}
                        flip={null}
                        icon={
                                    Array [
                                                "far",
                                                "check-circle",
                                              ]
                        }
                        listItem={false}
                        mask={null}
                        name=""
                        pull={null}
                        pulse={false}
                        rotation={null}
                        size={null}
                        spin={false}
                        symbol={false}
                        transform={null}
            />
            Bitpoll
</h1>,
          <SignIn
            login={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <FontAwesomeIcon
                border={false}
                className="logo-icon"
                fixedWidth={false}
                flip={null}
                icon={
                                Array [
                                                "far",
                                                "check-circle",
                                              ]
                }
                listItem={false}
                mask={null}
                name=""
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                transform={null}
/>,
              "Bitpoll",
            ],
            "className": "Header-title",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "border": false,
                "className": "logo-icon",
                "fixedWidth": false,
                "flip": null,
                "icon": Array [
                  "far",
                  "check-circle",
                ],
                "listItem": false,
                "mask": null,
                "name": "",
                "pull": null,
                "pulse": false,
                "rotation": null,
                "size": null,
                "spin": false,
                "symbol": false,
                "transform": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "Bitpoll",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "login": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Header matches snapshot with userId set 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    contributors={Object {}}
    displayName="Chase"
    getContributors={[Function]}
    isContributor={false}
    storeLogOut={[Function]}
    storeUser={[Function]}
    userId="10532835"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="Header-title"
>
          <FontAwesomeIcon
                    border={false}
                    className="logo-icon"
                    fixedWidth={false}
                    flip={null}
                    icon={
                              Array [
                                        "far",
                                        "check-circle",
                                      ]
                    }
                    listItem={false}
                    mask={null}
                    name=""
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    transform={null}
          />
          Bitpoll
</h1>,
        <Connect(Component)
          logOut={[Function]}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <FontAwesomeIcon
              border={false}
              className="logo-icon"
              fixedWidth={false}
              flip={null}
              icon={
                            Array [
                                          "far",
                                          "check-circle",
                                        ]
              }
              listItem={false}
              mask={null}
              name=""
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              transform={null}
/>,
            "Bitpoll",
          ],
          "className": "Header-title",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "border": false,
              "className": "logo-icon",
              "fixedWidth": false,
              "flip": null,
              "icon": Array [
                "far",
                "check-circle",
              ],
              "listItem": false,
              "mask": null,
              "name": "",
              "pull": null,
              "pulse": false,
              "rotation": null,
              "size": null,
              "spin": false,
              "symbol": false,
              "transform": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "Bitpoll",
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "logOut": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="Header-title"
>
            <FontAwesomeIcon
                        border={false}
                        className="logo-icon"
                        fixedWidth={false}
                        flip={null}
                        icon={
                                    Array [
                                                "far",
                                                "check-circle",
                                              ]
                        }
                        listItem={false}
                        mask={null}
                        name=""
                        pull={null}
                        pulse={false}
                        rotation={null}
                        size={null}
                        spin={false}
                        symbol={false}
                        transform={null}
            />
            Bitpoll
</h1>,
          <Connect(Component)
            logOut={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <FontAwesomeIcon
                border={false}
                className="logo-icon"
                fixedWidth={false}
                flip={null}
                icon={
                                Array [
                                                "far",
                                                "check-circle",
                                              ]
                }
                listItem={false}
                mask={null}
                name=""
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                transform={null}
/>,
              "Bitpoll",
            ],
            "className": "Header-title",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "border": false,
                "className": "logo-icon",
                "fixedWidth": false,
                "flip": null,
                "icon": Array [
                  "far",
                  "check-circle",
                ],
                "listItem": false,
                "mask": null,
                "name": "",
                "pull": null,
                "pulse": false,
                "rotation": null,
                "size": null,
                "spin": false,
                "symbol": false,
                "transform": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "Bitpoll",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "logOut": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
